d3a_interface/utils.py

```diff
@@ -253,3 +256,21 @@ def iterate_over_all_modules(modules_path):
         else:
             module_list.append(module_name)
     return module_list
+
+
+def deep_size_of(input_obj):
+    """
+    Gets the real size of a python object taking into consideration the nested objects
+    """
+    memory_size = 0
+    ids = set()
+    objects = [input_obj]
+    while objects:
+        new = []
```

Comment:
Minor, but how about renaming `new` to something like `child_objects` ?
AFAIU, these are the child objects, e.g. an `Offer` object inside a dict tree, right?

Reply:
@hannesdiedrich actually they are the referents objects not the direct children ,
we can name the new to referents_objs if you agree

Reply:
Thanks! yes, would be good.